### PR TITLE
[WIP] Add strategies for creating IP addresses and networks

### PR DESCRIPTION
**Pull request by alexwlchan**
This is a WIP pull request for adding strategies for IP address/networks, as discussed in issue #267.

I’ve got the code for creating IP addresses in. I think those strategies are nearly done, apart from review comments and additional bugs.

Still to do:

* [ ] IP networks
* [ ] Documentation
* [ ] Proper tests
* [ ] Do the work for setuptools extra, so that Python 2 users get the `ipaddress` backport by running `pip install hypothesis[ipaddress]`, a la Django.
* [ ] Sort out formatting -- CI is failing because I haven't run the linters et al. yet

### CODE REVIEW

File: scripts/enforce_header.py [hunk 1]
```diff
@@ -2,9 +2,9 @@
 import os
 import sys
 
-HEADER_FILE = "scripts/header.py"
+# HEADER_FILE = "scripts/header.py"
 
-HEADER_SOURCE = open(HEADER_FILE).read().strip()
+# HEADER_SOURCE = open(HEADER_FILE).read().strip()
```
Comment (alexwlchan) [inline scripts/enforce_header.py]: Gah, this is what happens when I'm careless and don't double-check what I'm staging.

**Comment by alexwlchan** [comment]
That will teach me to muck around with git rebase – I just blew all the code out of my branch, oops.

**Comment by alexwlchan** [comment]
* [ ] Apparently some people still use Python 2. I should probably test that.

**Comment by DRMacIver** [comment]
Note BTW that I've made the decision that as of Hypothesis 3.0 I'm fine adding backports as dependencies. So if you use a backport from the Python 3 standard library you don't need to add this as an extra.

The ongoing use of Python 2 is a surprise to me too.

**Comment by DRMacIver** [comment]
Also, sorry I haven't provided more feedback on this! I am keen, I've just been a bit distracted recently.

**Comment by alexwlchan** [comment]
Heh, no worries. I haven’t done too much on this either – work’s been pretty hectic recently.

Congrats on shipping Hypothesis 3.0!

**Comment by Daenyth** [comment]
For contrast you might like to look at the fake-factory integration, which has ipv4/ipv6 strategies.

**Comment by alexwlchan** [comment]
Dang, I dropped the ball on this, didn’t I?

I’m happy to pick it back up again and get it finished, but I suspect it will drop out quite neatly from the work to generate strings from arbitrary grammars. Assuming that’s the case (and that all this will thus be ripped out shortly), should we just close this for now?

**Comment by alexwlchan** [comment]
So I think this will neatly fall out of the work to generate strings from arbitrary patterns mentioned in the comments on #335.

My bandwidth to work on this sort of stuff just became quite limited, so for now I’m going to park this one – we can always reopen it in future.

**Pull request closed without merging**